cmd/mesh/manifest.go

```diff
@@ -89,6 +93,27 @@ func genManifests(inFilename string, setOverlayYAML string) (name.ManifestMap, e
 	if err != nil {
 		return nil, err
 	}
+	// for installation path contains version number only
+	isp := mergedICPS.InstallPackagePath
+	if version.IsVersionString(isp) {
+		mergedICPS.InstallPackagePath = fmt.Sprintf(helm.InstallationPathTemplate, isp, isp)
+	}
+	if util.IsHTTPURL(mergedICPS.InstallPackagePath) {
+		uf, err := helm.NewURLFetcher(mergedICPS.InstallPackagePath, "")
+		if err != nil {
+			return nil, err
+		}
+		errs := uf.FetchBundles()
+		if len(errs) != 0 {
+			return nil, errs.ToError()
+		}
+		isp := path.Base(mergedICPS.InstallPackagePath)
+		// get rid of the suffix, installation package is untared to folder name istio-{version}, e.g. istio-1.3.0
+		idx := strings.LastIndex(isp, "-")
+
+		mergedICPS.InstallPackagePath = filepath.Join(uf.GetDestDir(), isp[:idx], helm.ChartsFilePath)
```

Comment:
we should not be mutating mergedICPS - i can see why you did it, but maybe leave a TODO to fix that.

Reply:
done